mesh/submesh/pncsubmesh.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2010-2024, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "../../config/config.hpp"
+
+#ifdef MFEM_USE_MPI
+
+#include "pncsubmesh.hpp"
+
+#include <numeric>
+#include <unordered_map>
+#include "submesh_utils.hpp"
+#include "psubmesh.hpp"
+namespace mfem
+{
+
+using namespace SubMeshUtils;
+
+
+ParNCSubMesh::ParNCSubMesh(ParSubMesh& submesh,
+                           const ParNCMesh &parent, From from, const Array<int> &attributes)
```

Comment:
A rare long line. Thanks for correcting many other long lines in the `master` branch.